react/components/EditorContainer/StoreEditor/Styles/StyleEditor/typography/TypeToken.tsx

```diff
@@ -0,0 +1,189 @@
+      <StyleEditorHeader
+        title={prettify(token)}
+        buttonLabel={saveLabel}
+        onButtonClick={saveStyle}
+      />
+      <div className="ph6">
+        {Object.keys(font)
+          .filter(key => !key.startsWith('__'))
+          .map(key => (
+            <TypeTokenEntry
+              font={font}
+              dispatch={dispatch}
+              fontFamilies={data.listFonts}
+              id={key as keyof Font}
+            />
+          ))}
+      </div>
+    </>
+  )
+}
+
+function getOnChange(key: keyof Font, dispatch: Dispatch<Partial<Font>>) {
+  return (_: React.ChangeEvent, value: string) => dispatch({ [key]: value })
+}
+
+interface DropdownOption {
+  value: string
+  label: string
+}
+
+function getOptions(
+  key: keyof Font,
+  font: Font,
+  fontFamilies: FontFamily[]
+): DropdownOption[] {
+  const options = (() => {
+    switch (key) {
+      case 'fontFamily':
+        return fontFamilies.map(({ id, fontFamily }) => ({
+          label: fontFamily,
+          value: fontFamily,
+        }))
+      case 'fontWeight':
+        return [
+          { label: 'Thin', value: '100' },
+          { label: 'Extra Light', value: '200' },
+          { label: 'Light', value: '300' },
+          { label: 'Normal', value: '400' },
+          { label: 'Medium', value: '500' },
+          { label: 'Semi Bold', value: '600' },
+          { label: 'Bold', value: '700' },
+          { label: 'Extra Bold', value: '800' },
+          { label: 'Black', value: '900' },
+        ]
+      case 'fontSize':
+        return [
+          { label: '48px', value: '3rem' },
+          { label: '36px', value: '2.25rem' },
+          { label: '24px', value: '1.5rem' },
+          { label: '20px', value: '1.25rem' },
+          { label: '16px', value: '1rem' },
+          { label: '14px', value: '.875rem' },
+          { label: '12px', value: '.75rem' },
+        ]
+      case 'letterSpacing':
+        return [
+          { label: 'Normal', value: 'normal' },
+          { label: 'Tracked', value: '.1em' },
+          { label: 'Tracked Tight', value: '-.05em' },
+          { label: 'Tracked Mega', value: '.25em' },
+          { label: 'Zero', value: '0' },
+        ]
+      case 'textTransform':
+        return [
+          { label: 'None', value: 'none' },
+          { label: 'Capitalize', value: 'capitalize' },
+          { label: 'Uppercase', value: 'uppercase' },
+          { label: 'Lowercase', value: 'lowercase' },
+        ]
+    }
+  })()
+
+  const currentIfNotInOptions =
+    options.find(({ value }) => value === font[key]) == null
+      ? [{ label: prettify(font[key]), value: font[key] }]
+      : []
+
+  return [...currentIfNotInOptions, ...options]
+}
+
+function getValue(key: keyof Font, font: Font): string {
+  return font[key]
+}
+
+interface EntryProps {
+  font: Font
+  fontFamilies: FontFamily[]
+  id: keyof Font
+  dispatch: Dispatch<Partial<Font>>
+}
+
+const TypeTokenEntry: React.FunctionComponent<EntryProps> = ({
```

Comment:
Do you think moving this component to another file would make the code more organized?